boxoffice/forms/item.py

```diff
@@ -72,7 +74,16 @@ class ItemForm(forms.Form):
         validators=[forms.validators.DataRequired(__("Please specify the quantity available for sale"))])
     assignee_details = forms.TextAreaField(__("Assignee details"), filters=[format_json],
         validators=[validate_json], default=ASSIGNEE_DETAILS_PLACEHOLDER)
+    event_date = forms.DateField(__("Event date"), validators=[forms.validators.DataRequired(__("Please specify a date for the event"))])
     cancellable_until = forms.DateTimeField(__("Cancellable until"), validators=[forms.validators.Optional()])
+    place_supply_state_code = forms.SelectField(__("State"),
+        choices=[(state['short_code_text'], state['name']) for state in sorted(indian_states, key=lambda k: k['name'])],
+        description=__("Place of supply"),
+        default='KA')
+    place_supply_country_code = forms.SelectField(__("Country"),
+        choices=[(country.alpha_2, country.name) for country in sorted(pycountry.countries, key=lambda k: k.name)],
+        description=__("Place of supply"),
+        default='IN')
```

Comment:
We just did this in Funnel, where the conclusion was to move the code to Baseframe the next time it is required. Might as well do that now. @iambibhas @vidya-ram 

https://github.com/hasgeek/funnel/pull/367

Reply:
What is the source for state codes? In India, some states have different codes for license plates and taxation. Please ensure the source list has the correct code.

Reply:
Notes here. Please check what codes GST uses. https://en.wikipedia.org/wiki/ISO_3166-2:IN

Reply:
The baseframe PR can't be merged because of some issue with pycountry's locale. I'm trying to figure it out. I'll fix it and then migrate both funnel and boxoffice. For now, gonna merge this.

Reply:
ref: https://github.com/hasgeek/baseframe/pull/216